### PR TITLE
fix: sync when starting up language server in a multi-project setup

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

Previously, the initialization of language server was triggered several times.

This PR adds 
- asserts that 
  - content roots are only added after initialization
  - initialization is not occurring recursively
- retry mechanism to add a project to language server waiting up to 30s when starting up
- a fix when initializing with an API error on the sastEnabled call
- logging when shutting down

### Checklist

- [ ] Tests added and all succeed
- [x] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt [hunk 1]
```diff
@@ -81,23 +81,29 @@ class SnykTaskQueueService(val project: Project) {
     }
 
     fun connectProjectToLanguageServer(project: Project) {
-        synchronized(LanguageServerWrapper) {
```
Comment (teodora-sandu) [inline src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt:84]: Question: Why did you remove the `synchronized` block here? Is it because you bound the service to a project and so this function won't be running in parallel anymore?

Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt:84]: It was not needed, didn't bring the wanted effect, and it blocks usage of the class, so I removed it to have one possible deadlock less.

File: src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt [hunk 2]
```diff
@@ -41,7 +41,7 @@ import snyk.common.SnykError
 import snyk.common.lsp.LanguageServerWrapper
 import snyk.trust.confirmScanningAndSetWorkspaceTrustedStateIfNeeded
 
-@Service
+@Service(Service.Level.PROJECT)
```
[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt:44]: TaskQueueService was always project level, I think. It was just not specified like that, but relying on a project anyways.

[the remark this answers is not in the document]
Comment (teodora-sandu) [reply src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt:44]: I thought services were by default application level, if not specified: https://plugins.jetbrains.com/docs/intellij/plugin-services.html#examples. Does this change anything?

[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt:44]: It should not, as it cannot work without project. So all access and instantiation should be requiring a project.

[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt:44]: The usual access is using this: `fun getSnykTaskQueueService(project: Project): SnykTaskQueueService? = project.serviceIfNotDisposed()`

[the remark this answers is not in the document]
Comment (teodora-sandu) [reply src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt:44]: Ah okay thanks for explaining!

**Review by teodora-sandu — APPROVED** [review]
Tested it with 3 projects at once, like you said. It seems to work and the code makes sense!
